src/execution/utils.ts

```diff
@@ -162,9 +179,60 @@ async function signTransaction(
   }
 }
 
+async function signAuthorizations(
+  config: RhinestoneAccountConfig,
+  preparedTransaction: PreparedTransactionData,
+) {
+  return await signAuthorizationsInternal(config, preparedTransaction.data)
+}
+
+async function signAuthorizationsInternal(
+  config: RhinestoneAccountConfig,
+  data: IntentData | UserOpData,
+) {
+  const eoa = config.eoa
+  if (!eoa) {
+    throw new Error('EIP-7702 initialization is required for EOA accounts')
+  }
+  const accountAddress = getAddress(config)
+  const requiredDelegations =
+    data.type === 'intent'
+      ? data.intentRoute.intentOp.signedMetadata.account.requiredDelegations ||
+        {}
+      : {}
+  const authorizations: SignedAuthorization[] = []
+  for (const chainId in requiredDelegations) {
+    const delegation = requiredDelegations[chainId]
+    const chain = getChainById(Number(chainId))
+    if (!chain) {
+      throw new Error(`Chain not supported: ${chainId}`)
+    }
+    const walletClient = createWalletClient({
+      chain,
+      account: eoa,
+      transport: createTransport(chain, config.provider),
+    }).extend(publicActions)
+    const code = await walletClient.getCode({
+      address: accountAddress,
+    })
+    const isDelegated =
+      code === concat(['0xef0100', delegation.contract.toLowerCase() as Hex])
```

Comment:
will code always be lowercased?

Reply:
`code` coming from viem is always lowercased. `contract` coming from the API is checksum cased